src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -490,8 +518,36 @@ internal bool TryGetValue(int index, out DateTimeOffset value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (JsonHelpers.TryParseAsISO(segment, out DateTimeOffset tmp, out int bytesConsumed) &&
-                segment.Length == bytesConsumed)
+            DateTimeOffset tmp;
+            int bytesConsumed;
+
+            // Segment needs to be unescaped
+            if (row.HasComplexChildren)
+            {
+                int backslash = segment.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(backslash != -1);
+
+                Span<byte> segmentUnescaped = (segment.Length <= JsonConstants.StackallocThreshold)
+                    ? stackalloc byte[segment.Length]
+                    : new byte[segment.Length];
```

Comment:
Same here. Consider using array pool rather than allocating a new array or constrain the input size so it "fits" on the stack (like guid).